lib/game/game.dart

```diff
@@ -229,7 +229,7 @@ abstract class BaseGame extends Game with TapDetector {
   /// You can override it further to add more custom behaviour.
   @override
   void update(double t) {
-    components.addAll(_addLater);
+    _addLater.forEach(add);
```

Comment:
this doesn't look right to me. preAdd is called immediately when the component is added to the add later list. you are calling it twice by calling add here. I separated all handling to `preAdd` so we didn't have this problem.

Reply:
This makes sense, I don't know how I missed the `predAdd` method...